src/main/java/org/apache/sling/feature/analyser/Analyser.java

```diff
@@ -236,29 +244,80 @@ public BundleDescriptor getFrameworkDescriptor() {
 
                 @Override
                 public void reportWarning(final String message) {
-                    warnings.add(message);
+                    if (analyserMetaDataExtension != null && analyserMetaDataExtension.reportWarning(feature.getId())) {
```

Comment:
Why must analyserMetaDataExtension  be available to report anything?

Reply:
@cziegeler, good catch! This was supposed to be the other way around - I fixed it.